### PR TITLE
Enhancement/remove name field services

### LINKED ISSUE #31: Remove name field of the Services' definitions at the yaml

### PR DESCRIPTION
**Pull request by lucaspeixotot**
Closes #31 

Signed-off-by: Lucas Peixoto <lucaspeixotoac@gmail.com>

### CODE REVIEW

File: zeta/zeta.py [hunk 1]
```diff
@@ -155,19 +160,22 @@ def gen_creation(self):
...
                 if 'persistent' in v and v['persistent']:
                     persistent = "1"
+
                 # Getting callbacks
                 if 'subscribers' in v:
                     subscribers_list = list()
                     for s in v['subscribers']:
-                        subscribers_list.append(s['name'] +
-                                                "_service_callback")
+                        for s_k, s_v in s.items() :
```
Comment (rodrigopex) [inline zeta/zeta.py:168]: If you will not use the value here, it seems to be better to use .keys() instead of .items()

File: zeta/zeta.py [hunk 2]
```diff
@@ -155,19 +160,22 @@ def gen_creation(self):
...
+                                                    "_service_callback")
                     subscribers_init = "zt_callback_f " + name_subscribers + "[] = { " + ", ".join(
                         subscribers_list) + ", NULL };"
                     subscribers = name_subscribers
                 if 'publishers' in v:
                     publishers_list = list()
                     for p in v['publishers']:
-                        publishers_list.append(p['name'] + "_thread_id")
+                        for p_k, p_v in p.items() :
+                            publishers_list.append(p_k + "_thread_id")
```
Comment (rodrigopex) [inline zeta/zeta.py:178]: Same here

File: zeta/zeta.py [hunk 3]
```diff
@@ -252,9 +260,9 @@ def __init__(self, yaml_dict):
     def create_substitutions(self):
         for s in self.services:
             for k, v in s.items():
-                name = v['name']
-                name_tid = v['name'] + "_thread_id"
-                name_thread = v['name'] + "_task"
+                name = k
+                name_tid = k + "_thread_id"
+                name_thread = k + "_task"
```
Comment (rodrigopex) [inline zeta/zeta.py:265]: I would suggest an fString instead. For example: 
```python
name_tid = f"{k}_thread_id"
name_thread = f"{k}_task"
```

File: zeta/zeta.py [hunk 4]
```diff
@@ -277,9 +285,9 @@ def __init__(self, yaml_dict):
...
```
Comment (rodrigopex) [inline zeta/zeta.py:290]: Same here

File: zeta/zeta.py [hunk 5]
```diff
@@ -16,6 +16,11 @@
 ZETA_SRC_DIR = "."
 ZETA_INCLUDE_DIR = "."
 
+class YamlRefLoader(yaml.SafeLoader) :
+    def __init__(self, stream) :
+        super(YamlRefLoader, self).__init__(stream)
+    def ref(self, node):
+        return self.construct_scalar(node)
```
Comment (rodrigopex) [inline zeta/zeta.py:23]: I could not see any warning or error to the user when something is wrong. Is there anything related to it?

Comment (lucaspeixotot) [reply zeta/zeta.py:23]: I think that there isn't. Ref function will be called just when some "!ref" prefix appears in YAML file, and your responsibility is just remove "!ref" text to ZetaCLI deal with it later. If some warning or error must be put about it I think that the right location would be in construct_yaml function. We can discuss about that.

Comment (rodrigopex) [reply zeta/zeta.py:23]: Ok. Let's talk about this later. :+1:

**Pull request merged**